.github/workflows/ci.yml

```diff
@@ -89,81 +93,168 @@ jobs:
-            image: "ubuntu:22.04"
-            suite: jammy
-            install: "make ca-certificates git curl golang"
-          - name: Debian-10
-            image: "debian:buster"
-            suite: buster
-            GOVER: '1.21'
-            install: "make ca-certificates git curl"
-          - name: Debian-11
-            image: "debian:bullseye"
-            suite: bullseye
-            GOVER: '1.21'
-            install: "make ca-certificates git curl"
-          - name: Debian-12
-            image: "debian:bookworm"
+          - name: "Debian 13/testing"
+            suite: trixie
+            image: debian:trixie-slim
+          - name: "Debian 12/bookworm"
             suite: bookworm
-            install: "make ca-certificates git curl golang"
+            image: debian:bookworm-slim
+          - name: "Debian 11/bullseye"
+            suite: bullseye
+            image: debian:bullseye-slim
+          - name: "Debian 10/buster"
+            suite: buster
+            image: debian:buster-slim
+          - name: "Ubuntu 24.04"
+            suite: noble
+            image: ubuntu:24.04
+          - name: "Ubuntu 22.04"
+            suite: jammy
+            image: ubuntu:22.04
+          - name: "Ubuntu 20.04"
+            suite: focal
+            image: ubuntu:20.04
     container:
       image: ${{ matrix.image }}
       env:
         APT_LISTCHANGES_FRONTEND: none
         DEBIAN_FRONTEND: noninteractive
     steps:
-      - name: Install O/S packages
+      - name: "Install packages"
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends ${{ matrix.install }}
+          apt-get install -y --no-install-recommends make ca-certificates git curl build-essential devscripts dh-golang binutils-i686-linux-gnu binutils-aarch64-linux-gnu binutils-arm-linux-gnueabihf jq
           git config --global --add safe.directory "$GITHUB_WORKSPACE"
-          echo GOVER: ${{ env.GOVER }}
 
-      - name: Checkout repository
+      - name: "Checkout repository"
         uses: actions/checkout@v3
         with:
-          # fetch the whole repo for `git describe` to
-          # work and get the nightly verion
+          # fetch the whole repo for `git describe` to work
           fetch-depth: 0
 
-      - name: Setup Go
-        if: ${{ matrix.GOVER != '' }}
+      - name: "Read go version from go.mod"
+        run: |
+          echo "GOVER=$(sed -n 's/^go \(.*\)/\1/p' go.mod)" >> $GITHUB_OUTPUT
+        id: goversion
+
+      - name: "Setup Go"
         uses: actions/setup-go@v3
         with:
-          go-version: ${{ matrix.GOVER }}
+          go-version: ${{ steps.goversion.outputs.GOVER }}
+
+      - name: Check is CI or Release
+        run: |
+          if [ -z "`git tag --points-at HEAD`" ]; then
+            releasetype=ci
+          else
+            releasetype=release
+          fi
+          echo RELEASETYPE=$releasetype >> $GITHUB_OUTPUT
+        id: releasetype
```

Comment:
this should probably also be moved to the Makefile, can also be merged with the `RELEASE` variable I added there for a similar purpose